### ISSUE TITLE
Support for Azure Firewall SNAT private ranges

### ISSUE THREAD

**Issue by jaygridley** [terraform-providers/terraform-provider-azurerm#7504]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Please add support for setting Azure Firewall SNAT private ranges, see https://docs.microsoft.com/en-us/azure/firewall/snat-private-range. API should be there, see https://docs.microsoft.com/en-us/cli/azure/ext/azure-firewall/network/firewall?view=azure-cli-latest#ext-azure-firewall-az-network-firewall-update.

### New or Affected Resource(s)

* azurerm_firewall

**Comment by magodo** [comment]
Thank you for letting us know this issue :+1: I'll take a look and submit a PR to support his soon, hopefully!

**Comment by stefan-caraiman** [comment]
Any updates on this issue @magodo ?

**Comment by magodo** [comment]
@stefan-caraiman Unfortunately, this is currently blocked by the Azure Swagger spec, since that private range property is defined in `additionalProperties` map, which is not ideal for us to bring into Terraform codebase, given it might breakingly change.

**Comment by jaygridley** [comment]
@magodo The same was done with DNS proxy support in https://github.com/terraform-providers/terraform-provider-azurerm/pull/8878 and was merged. Can we revisit this?

**Comment by johnvanhienen** [comment]
Any updates @magodo ? It can be a solution for this issue if it was accepted in #8878 . It's one of the few items I need to use powershell to configure this property, and would prefer to do it with terraform

**Comment by magodo** [comment]
@jaygridley , @johnvanhienen I've opened a new PR for this (#10627).
